client/main/dataset.js

```diff
@@ -455,10 +456,20 @@ const DataSetModel = Backbone.Model.extend({
             levels = new Array(columnPB.levels.length);
             for (let i = 0; i < levels.length; i++) {
                 let levelPB = columnPB.levels[i];
-                levels[i] = {
-                    label: levelPB.label,
-                    value: levelPB.value,
-                };
+                if (column.measureType === 'nominaltext') {
+                    levels[i] = {
+                        label: levelPB.label,
+                        value: i,
+                        importValue: levelPB.importValue
+                    };
+                }
+                else {
+                    levels[i] = {
+                        label: levelPB.label,
+                        value: levelPB.value,
+                        importValue: levelPB.value.toString()
```

Comment:
I don't see the advantage of not having it here. If it's not calculated here the conversion and check will be needed everywhere else that importValue is used. The tiny bit of memory that it uses on the client side is not that big of an issue to not have to think about it again.